cclunit-framework-source/src/main/resources/cclutmock_table.inc

```diff
@@ -0,0 +1,728 @@
+@param rowData
+    A pipe-delimited string of data to be inserted into the table.
+
+Example:
+call cclutDefineMockTable("person", "person_id|name_last|name_first|birth_dt_tm", "f8|vc|vc|dq8")
+call cclutCreateMockTable("person")
+call cclutAddMockData("person", "1.0|Washington|George|01-JAN-1970 00:00") ;Will add George Washington
+call cclutAddMockData("person", "2.0|A\|d\\ams|John|02-FEB-1971 11:11") ;Will add John A|d\ams
+call cclutAddMockData("person", "3.0|Jefferson|\null|03-MAR-1972 22:22") ;Will add Jefferson (no first name)
+call cclutAddMockData("person", "4.0|Madison||04-APR-1973 10:33") ;Will add Madison (empty string for first name)
+**/
+subroutine cclutAddMockData(tableName, rowData)
+    ;Declare local variables
+    declare CCLUT_TABLE_NAME = vc with protect, constant(trim(cnvtupper(tableName), 3))
+    declare CCLUT_ROW_DATA = vc with protect, constant(rowData)
+    declare cclutMockTablePos = i4 with protect, noconstant(0)
+    declare cclutMockTableIndex = i4 with protect, noconstant(0)
+    declare cclutMockDataLength = i4 with protect, noconstant(size(CCLUT_ROW_DATA))
+    declare cclutMockDataIndex = i4 with protect, noconstant(0)
+    declare cclutMockColumnLength = i4 with protect,
+        noconstant(size(cclut_mockTables->tables[cclutMockTablePos].columns, 5))
+    declare cclutMockColumnIndex = i4 with protect, noconstant(1)
+    declare cclutHasComma = i4 with protect, noconstant(0)
+    declare cclutNullColumn = i4 with protect, noconstant(0)
+    declare cclutDataBuffer = vc with protect, noconstant(" ")
+    declare cclutEscapeBuffer = vc with protect, noconstant("")
+    declare cclutDataCharacter = vc with protect, noconstant("")
+    declare cclutParserText = vc with protect, noconstant("")
+
+    ;Validate inbound parameters
+    if (CCLUT_TABLE_NAME <= " ")
+        call cclexception(100, "E", "cclutAddMockData() - tableName is a required field")
+        return
+    endif
+
+    ;Validate that the table exists and has been created
+    set cclutMockTablePos = locateval(cclutMockTableIndex, 1, size(cclut_mockTables->tables, 5), CCLUT_TABLE_NAME,
+        cclut_mockTables->tables[cclutMockTableIndex].originalName)
+    if (cclutMockTablePos < 1)
+        call cclexception(100, "E", concat("cclutAddMockData() - tableName has not been mocked. tableName: ",
+            CCLUT_TABLE_NAME))
+        return
+    elseif (cclut_mockTables->tables[cclutMockTablePos].isFinalized = FALSE)
+        call cclexception(100, "E", concat("cclutAddMockData() - tableName has not been created. tableName: ",
+            CCLUT_TABLE_NAME))
+        return
+    endif
+
+    ;Initialize insert statement
+    set cclutParserText = concat(" insert into ", cclut_mockTables->tables[cclutMockTablePos].replaceName, " set ")
+    call parser(cclutParserText)
+
+    ;Special case if rowData is empty for single-column tables
+    if (cclutMockDataLength = 0 or (cclutMockDataLength = 1 and ichar(substring(1, 1, CCLUT_ROW_DATA)) = 0))
```

Comment:
So, I think I should change my test for this scenario to explicitly do textlen/ichar instead of just a string assertion to avoid any of the string shenanigans in the language itself.  Having said that, I agree with you, but I want to make sure I understand your test cases, because they don't seem quite right.

For textlen/ichar respectively,
char(0)/"" = 1/0 (expected, as you mentioned)
notrim(" ") = 1/32 (expected, as you mentioned assuming this is what you meant by " ")
trim(" ")/trim("") = 0/32 (this is what I'd expect... well, the 0 textlen at least, I'll assume the 32 ichar is another quirk of CCL; but you said you expect it to be 1/32)

Can you elaborate more on why you feel it should be 1/32?  I tested it on a raw table (outside the context of my framework) and also got 0/32 (which is what I'd expect).
![image](https://user-images.githubusercontent.com/44979171/51406426-b9b31180-1b1e-11e9-9cd1-27c2eca5f7ba.png)

Reply:
So how are you wanting to handle it then?  I had specifically uplifted the code to handle scenarios like that here:
https://github.com/cerner/cclunit-framework/pull/7/files#diff-442f2b23a77dabc96116c7ad8270889cR711

My thought is that if the user creates a trim(" ") vc and does a textlen/ichar on the vc, they will get 0/32.  Therefore, we should insert the data as such into the database that if they read it out, it's 0/32 for consistency.  I can change it to be 1/32 if you'd like, but then, I _believe_ it would be impossible to insert 0 (true empty) using the API, which doesn't seem desirable.

Reply:
The point of my last response was merely to provide the requested elaboration how I got 1/32 rather than 0/32.  I find it fascinating that CCL says that variables set equal to trim("") and trim(" ") are equal to each other but returns 0/0 for the first and 0/32 for the second.  On the other hand, if the length mystr is zero then ichar(mystr) isn't really defined so it hardly matters what value ichar returns in that case.

My original concern was that "empty", i.e., trim("") having length zero and "ought to be empty", i.e., "" having length one were being conflated (for a single field table anyway). That seems to be addressed now.  If a test author really wants "" inserted, s/he just needs to put a char(0) between the pipes.

Reply:
Yeah, I'm not too worried about the ichar for the reason you mentioned; I was just including it for your original comparison.

That all sounds good though; I'll leave it how it is now.